agent_docs/architecture.md

```
@@ -0,0 +1,130 @@
+# Architecture & Patterns
+
+## Backend Structure
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)